jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -266,139 +265,144 @@ public String toString()
         return String.format("%s@%x{s=%d/%d,p=%d}",
             getClass().getSimpleName(),
             hashCode(),
-            _size.get(),
+            _count.getLo(),
             _capacity,
-            _pending.get());
+            _count.getHi());
     }
 
-    private class ReservedThread implements Runnable
+    private enum State
     {
-        private final SynchronousQueue<Runnable> _task = new SynchronousQueue<>();
-        private boolean _starting = true;
-
-        public boolean offer(Runnable task)
-        {
-            if (LOG.isDebugEnabled())
-                LOG.debug("{} offer {}", this, task);
-
-            try
-            {
-                _task.put(task);
-                return true;
-            }
-            catch (Throwable e)
-            {
-                LOG.ignore(e);
-                _size.getAndIncrement();
-                _stack.offerFirst(this);
-                return false;
-            }
-        }
+        PENDING,
+        RESERVED,
+        RUNNING,
+        IDLE,
+        STOPPED
+    }
 
-        public void stop()
-        {
-            offer(STOP);
-        }
+    private class ReservedThread implements Runnable
+    {
+        // The state and thread are kept only for dumping
+        private volatile State _state = State.PENDING;
+        private Thread _thread;
```

Comment:
`_thread` should be marked `volatile` as `toString()` will be called from concurrent threads.